packages/nextjs-mf/src/plugins/container/InvertedContainerPlugin.ts

```diff
@@ -86,6 +86,113 @@ class InvertedContainerPlugin {
+                // modules.forEach(chunk.removeModule);
+                const getmodules = chunk.getModules();
+
+                console.log(getmodules.length, chunk.name);
+              }
+            }
+          }
+        );
+        compilation.hooks.optimizeChunkModules.tap(
+          'InvertedContainerPlugin',
+          (chunks, modules) => {
+            for (const chunk of chunks) {
+              if (
+                chunk.hasRuntime() &&
+                chunk.name === this.options?.container
+              ) {
+                const eagerModulesInRemote = new Set();
+
+                for (const module of modules) {
+                  if (!compilation.chunkGraph.isModuleInChunk(module, chunk)) {
+                    continue;
+                  }
+
+                  if (
+                    module.type === 'provide-module' ||
+                    module.type === 'consume-shared-module'
+                    // || module.constructor.name === 'NormalModule'
+                  ) {
+                    // if its a shared module and is eager, we want to remove that eager reference
+                    //@ts-ignore
+                    if (module._eager) {
+                      //@ts-ignore
+                      eagerModulesInRemote.add(module._request);
+                    }
+                    //     console.log(module);
+                    //@ts-ignore
+                    if (module?._eager || module?._name?.startsWith('next')) {
+                      const outgoing =
+                        compilation.moduleGraph.getOutgoingConnections(module);
+                      const incoming =
+                        compilation.moduleGraph.getIncomingConnections(module);
+
+                      compilation.chunkGraph.disconnectChunkAndModule(
+                        chunk,
+                        module
+                      );
+                    }
+                  }
+                }
+                for (const module of modules) {
+                  if (!compilation.chunkGraph.isModuleInChunk(module, chunk)) {
+                    continue;
+                  }
+                  if (
+                    // module.type === 'consume-shared-module'
+                    module.constructor.name === 'NormalModule'
+                  ) {
+                    // if its a shared module and is eager, we want to remove that eager reference
+                    //@ts-ignore
+                    if (eagerModulesInRemote.has(module.resource)) {
+                      // const outgoing =
+                      //   compilation.moduleGraph.getOutgoingConnections(module);
+                      // const incoming =
+                      //   compilation.moduleGraph.getIncomingConnections(module);
+                      const modulesToRemove: Set<Module> = new Set();
+                      modulesToRemove.add(module);
+                      module.dependencies.forEach((dep) => {
+                        modulesToRemove.add(
```

Comment:
get dependencies of shared module like react/index.js imports ./cjs/react.development.js - ensure all dependency trees are removed from container chunk.